src/App.js

```diff
@@ -1,61 +1,91 @@
 import React from 'react';
 
-function App() {
-  return (
-    <section className="todoapp">
-      <header className="header">
-        <h1>todos</h1>
-
-        <input
-          className="new-todo"
-          placeholder="What needs to be done?"
-          autoFocus=""
-        />
-      </header>
-
-      <section className="main" style={{ display: 'block' }}>
-        <input id="toggle-all" className="toggle-all" type="checkbox" />
-        <label htmlFor="toggle-all">Mark all as complete</label>
-        <ul className="todo-list">
-          <li className="">
-            <div className="view">
-              <input className="toggle" type="checkbox" />
-              <label>sdfsdfsdf</label>
-              <button className="destroy"></button>
-            </div>
-          </li>
-          <li className="">
-            <div className="view">
-              <input className="toggle" type="checkbox" />
-              <label>dsfgsdfgdsrg</label>
-              <button className="destroy"></button></div>
-          </li>
-          <li className="">
-            <div className="view">
-              <input className="toggle" type="checkbox" />
-              <label>sddfgdfgdf</label>
-              <button className="destroy"></button>
-            </div>
-          </li>
-        </ul>
+import ToDoItem from './ToDoItem'
+import Header from './Header'
+import Footer from './Footer'
+
+class App extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      toDoItems: [],
+      selectedFilter: "all"
+    }
+  }
+
+  componentDidMount() {
+    if (localStorage.getItem("toDoItems") !== null) {
+      this.setState({
+        toDoItems: JSON.parse(localStorage.getItem("toDoItems")),
+        selectedFilter: JSON.parse(localStorage.getItem("selectedFilter"))
+      })
+    }
+  }
+
+  saveDataToLocalStorage() {
+    localStorage.setItem('toDoItems', JSON.stringify(this.state.toDoItems))
+    localStorage.setItem('selectedFilter', JSON.stringify(this.state.selectedFilter))
+  }
+
+  updateToDoItems = (newItem) => {
+    this.setState({
+      toDoItems: [...this.state.toDoItems, newItem],
+    })
+  }
+
+  replaceItemWithChangedState = (itemToReplace) => {
+    const toDoArray = [...this.state.toDoItems];
+    const index = toDoArray.indexOf(itemToReplace);
+    itemToReplace.isCompleted = !itemToReplace.isCompleted;
```

Comment:
--- OBJECT MUTATION

Do you remember what happens when you try to assign object to the variable?
```javascript
const a = {
  name: 'Vasya'
  age: 21
}
const b = a;
const c = b;
b.age = 22;
c.name = 'Petya';
console.log(a); // {name: 'Petya', age: 22}
console.log(b); // {name: 'Petya', age: 22}
console.log(c); // {name: 'Petya', age: 22}
```
You should avoid this kind of code. Use `...spread` operator to create a copy of object and override needed fields.